lib/active_shipping/shipment_packer.rb

```diff
@@ -40,6 +40,7 @@ def pack(items, dimensions, maximum_weight, currency)
             items_to_pack.reject! { |i| i[:quantity].to_i == 0 }
             state = :package_full
           when :package_full
+            validate_package_quantity(packages.count)
```

Comment:
Is this valuable? If we get to this case, then the loop will terminate, and validating the quantity in general I figure is/was a way to put an upper bound on load limit on runtime.  When `:package_full` we will be doing no more work.  Unless we're concerned we may generate too big of a JSON payload?

Maybe we should be doing this during `:filling_package` instead?

I _think_ I might like to see this change as its own PR, since the bulk of this PR is renaming and cleaning up code, while this is introducing a new behaviour